what4/src/What4/Protocol/SMTWriter.hs

```diff
@@ -931,6 +947,19 @@ class (SupportTermOps (Term h)) => SMTWriter h where
                 -> Term h
                 -> Command h
 
+  synthFunCommand :: f h
+                  -> Text
+                  -> [(Text, Some TypeMap)]
+                  -> TypeMap tp
+                  -> Command h
+
+  declareVarCommand :: f h
+                    -> Text
+                    -> TypeMap tp
+                    -> Command h
+
+  constraintCommand :: f h -> Term h -> Command h
```

Comment:
These functions deserve some Haddocks, as downstream libraries will need to know how to implement these methods for their own `SMTWriter` instances.